plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -552,8 +552,8 @@ public boolean isLimitGuaranteed(ConnectorSession session)
     @Override
     public OptionalLong delete(ConnectorSession session, JdbcTableHandle handle)
     {
-        // ClickHouse does not support DELETE syntax, but is using custom: ALTER TABLE [db.]table [ON CLUSTER cluster] DELETE WHERE filter_expr
-        throw new TrinoException(NOT_SUPPORTED, MODIFYING_ROWS_MESSAGE);
+        checkArgument(!handle.getConstraint().isAll(), "Deleting without constraint is not supported: %s", handle);
```

Comment:
We don't need to disallow this condition because ClickHouse supports `WHERE true` condition.

Reply:
removed override

Reply:
@ebyhr jfyi: without this check in trino, the error returned to the user is not user friendly. (the error from clickhouse is propagated back to the trino user). it makes sense in the clickhouse context, but not as per the format of other trino errors.

Reply:
You need to modify DELETE query to include `WEHRE true` instead.

Reply:
@ebyhr - sorry i did not follow this. are you suggesting to add the   where true clause in the Clickhouse client if no condition is mentioned in the trino query?